modules/machine-config-overview.adoc

```diff
@@ -0,0 +1,101 @@
+// Module included in the following assemblies:
+//
+// * operators/operator-reference.adoc
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+[id="machine-config-overview{context}"]
+= MachineConfig overview
+
+The Machine Config Operator (MCO) lets you apply changes to
+the operating systems running on {product-title} nodes.
+Understanding what MCO does and how it interacts with other components
+is critical to making advanced, system-level changes to an
+{product-title} cluster. Here are some things you should know about MCO,
+MachineConfigs, and how they are used:
+
+* The most common way to use MCO is to create a MachineConfig that
+makes a specific change to a file or service on the operating system
+for each system representing a pool of {product-title} nodes.
+
+* MCO applies operating system changes to pools of machines. All {product-title} clusters
+start with bootstrap, worker, and master node pools. By adding more
+role labels, you can configure special pools of nodes. So, for example,
+you could set up special worker nodes that include particular hardware
+features needed by an application. However, examples in this section focus
+on changes to the default pool types.
+
+* Some per node configuration must be in place before {product-title} is
+is installed to disk. In most cases, this can be accomplished by creating
+a MachineConfig that is injected directly into the {product-title} installer
+process, instead of running as a post-installation MachineConfig.
+In other cases, you might need to do bare metal installation where you
+pass kernel arguments at {product-title} installer start-up, to do such
+things as set per-node individual IP addresses or advanced disk partitioning.
+
+* MCO manages items that are set in MachineConfigs. Manual changes you do to
+your systems will not be overwritten by MCO, unless MCO is explicitly told to
+manage a conflicting file. In other words, MCO only makes specific updates
+you request, it does not claim control over the whole node.
+
+* Direct changes to nodes are strongly discouraged. If you need to decommission
+a node and start a new one, those direct changes would be lost.
+
+* MCO can only be set to write to files in `/etc` and `/var` directories, although
+there are symbolic links to some directories that can be writeable by being
+symbolically linked to one of those areas. The `/opt` directory is an example.
+
+* Ignition is the configuration format used in MachineConfigs. See the
+link:https://github.com/coreos/ignition/blob/master/docs/configuration-v3_1.md[Ignition Configuration Specification v3.1.0]
+for details.
+
+* Although Ignition config settings can be delivered directly at {product-title}
+installation time, and are formatted in the same way that MCO delivers Ignition
+configs, MCO has no way of seeing what those original Ignition configs are. So, whenever
+possible, you should wrap Ignition config settings into a MachineConfig before
```

Comment:
Instead of "So, whenever possible, you should wrap...", I would either be more direct an say, "You must wrap...." For less direct, maybe, "It is recommended that you wrap...."